Student/Student/Routes.swift

```diff
@@ -22,7 +22,7 @@ import Core
 import TechDebt
 
 public let router: Router = {
-let routeMap: [String: RouteHandler.ViewFactory?] = [
+let routeMap: KeyValuePairs<String, RouteHandler.ViewFactory?> = [
```

Comment:
I was experiencing issues with the sorting that happens on `routeMap` -> `handlers` because:
`/courses/:courseID/pages/:url` would get sorted higher than:
`/:context/:contextID/pages/new` thus making it impossible to route to a route such as:
`/courses/1/pages/new` because it would always take me to the page details screen.

I think having explicit control over the ordering for routes is a win all around though as problems like these in routes are not infrequent.